src/theory/arith/nl/nonlinear_extension.cpp

```diff
@@ -400,12 +401,15 @@ int NonlinearExtension::checkLastCall(const std::vector<Node>& assertions,
                                       std::vector<NlLemma>& lems,
                                       std::vector<NlLemma>& wlems)
 {
-  // initialize the non-linear solver
-  d_nlSlv.initLastCall(assertions, false_asserts, xts);
-  // initialize the trancendental function solver
   std::vector<NlLemma> lemmas;
-  d_trSlv.initLastCall(assertions, false_asserts, xts, lemmas);
-
+  if (options::nlExt())
+  {
+    // initialize the non-linear solver
+    d_nlSlv.initLastCall(assertions, false_asserts, xts);
+    // initialize the trancendental function solver
+    d_trSlv.initLastCall(assertions, false_asserts, xts, lemmas);
+  }
+  
   // process lemmas that may have been generated by the transcendental solver
   filterLemmas(lemmas, lems);
```

Comment:
This could also be moved into the above if-block, if you didn't leave it there intentionally so that a new block for nl-cad can be inserted more easily.

Reply:
Yes, I'll move.